tests/_manim.py

```diff
@@ -205,15 +264,20 @@ def text2svg(self):
         width = 600
         height = 400
 
-        return text2svg(
-            settings,
-            size,
-            line_spacing,
-            disable_liga,
-            file_name,
-            30,
-            30,
-            width,
-            height,
-            self.text,
-        )
+        try:
+            return text2svg(
+                settings,
+                size,
+                line_spacing,
+                disable_liga,
+                file_name,
+                30,
+                30,
+                width,
+                height,
+                self.text,
+            )
+        except ValueError as e:
+            if os.path.exists(file_name):
+                os.remove(file_name)
```

Comment:
I think this will still error out. The error was not `FileNotExistsError` it was `PermissionError`(which mean the file was opened but wasn't closed). I guess this part of the code can be removed.

Reply:
The issue is that on my machine pango left an empty svg file when this occurred and it interferes with caching in. That's why I've added cleaning, but it seems windows works differently.

Reply:
I don't think there exist any caching for `tests/_manim.py`. Also, I think it should be fine to not remove the file at all since it's just used while testing.

Reply:
The caching problem was present when running `manim`. I'll remove it from `_manim` tests but it may cause problems later on `manim` repo with windows build.

Reply:
In the manim repo, you can check the number of submobjects in the parsed object. If the count is `0` it's an empty SVG(in which case delete the SVG) if not it's correct and  it cached.

Reply:
Empty svg files cannot be parsed. I get `ExpatError: no element found: line 1, column 0` when I try.

Reply:
Hmm, did you mean by empty SVG file a literal empty file? In my case when I tried previously (without the checks for valid colour), I got a valid SVG file but nothing was in it.

Reply:
I did meant literal empty file. Without validation it was a correct svg but now a `ValueError` is raised so `ManimPango` creates the file but doesn't populate it.

Reply:
> Without validation it was a correct svg but now a ValueError is raised so ManimPango creates the file but doesn't populate it.

Ah, I see what's wrong. We should close the file before erroring out(which should create a valid SVG) or check before even creating a file. That is we should move the checking part before calling `cairo_svg_surface_create` or we should say cairo to close the file handles by calling `cairo_destroy()`, `cairo_surface_destroy()`.

Reply:
You're right, that was missing. That's probably also why windows couldn't get permission to delete the file, because it was still owned by the program.